### PR TITLE
fix: enum values were raw [APE-1551]

### PR DESCRIPTION
**Pull request by antazoey**
### What I did

* Issue where all enums were values instead of enum types
* Prevents exceptions in `__repr__`
* Fix issues where plugin type was a str instead of enum
* add some tests
* fix issue where geth connection ID was not very identifiable

### Checklist

- [ ] All changes are completed
- [ ] New test cases have been added
- [ ] Documentation has been updated

### CODE REVIEW

File: src/ape_plugins/utils.py [hunk 1]
```diff
@@ -373,14 +373,27 @@ def __bool__(self) -> bool:
...
+            # Prevent exceptions happening in repr()
+            logger.log_debug_stack_trace()
+            return "<PluginGroup>"
 
     def __str__(self) -> str:
         return self.to_str()
 
+    @validator("plugin_type")
+    def validate_plugin_type(cls, value):
+        return PluginType(value) if isinstance(value, str) else value
```
Comment (mikeshultz) [inline src/ape_plugins/utils.py:388]: question: So was something not initializing with a valid enum?

**Review by mikeshultz — APPROVED** [review]
LGTM

Thanks for tracking that down.  Sure don't see Python segfaults very often!

**Comment by mikeshultz** [comment]
I jumped the gun on approval, I guess.  Tests sure don't seem happy about that change.

File: src/ape_plugins/utils.py (hunk 1 above)
Comment (antazoey) [reply src/ape_plugins/utils.py:388]: somehow yes but I don't see where! I can't find he instance where we do that anywhere in our code, that's why im quite confused... it seems like sometimes the `Enum` part doesnt work in pydantic.

**Comment by antazoey** [comment]
> I jumped the gun on approval, I guess. Tests sure don't seem happy about that change.

it should be good now

**Review by fubuloubu — APPROVED** [review]

**Comment by antazoey** [comment]
please merge ASAP

**Pull request merged**